workers/api-server/src/http.rs

```diff
@@ -175,6 +185,17 @@ pub(super) fn parse_task_id_from_params(
         .map_err(|_| bad_request(ERR_TASK_ID_PARSE.to_string()))
 }
 
+/// A helper to parse out a matching pool name from a URL param
+pub(super) fn parse_matching_pool_from_params(
+    params: &UrlParams,
+) -> Result<String, ApiServerError> {
+    params
+        .get(MATCHING_POOL_URL_PARAM)
+        .ok_or_else(|| bad_request(ERR_MATCHING_POOL_PARSE.to_string()))?
+        .parse()
+        .map_err(|_| bad_request(ERR_MATCHING_POOL_PARSE.to_string()))
```

Comment:
nit: you don't need to go `to_string` on this, `bad_request` will call `to_string`